src/libraries/System.Collections/tests/Generic/Comparers/ReferenceEqualityComparer.Tests.cs

```diff
@@ -0,0 +1,94 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Reflection;
+using System.Runtime.CompilerServices;
+using Xunit;
+
+namespace System.Collections.Generic.Tests
+{
+    public class ReferenceEqualityComparerTests
+    {
+        [Fact]
+        public void TypeHasNoPublicCtors()
+        {
+            ConstructorInfo[] ctors = typeof(ReferenceEqualityComparer).GetConstructors(BindingFlags.Instance | BindingFlags.Public | BindingFlags.NonPublic);
+            Assert.DoesNotContain(ctors, ctor => ctor.IsPublic || ctor.IsFamily || ctor.IsFamilyOrAssembly);
+        }
+
+        [Fact]
+        public void InstanceProperty_ReturnsSingleton()
+        {
+            ReferenceEqualityComparer comparer = ReferenceEqualityComparer.Instance;
+            Assert.NotNull(comparer);
+
+            // use reflection to call property getter and make sure it's the exact same instance
+            object comparerFromReflection = typeof(ReferenceEqualityComparer).GetProperty("Instance").GetValue(null);
```

Comment:
I'm curious why we use reflection... why not just:
```C#
Assert.Same(comparer, ReferenceEqualityComparer.Instance);
```
?

Reply:
It was more of a defensive check in case this property getter ever got intrinsicified. If I'm being overly paranoid I can rip out the reflection aspect. :)

Reply:
Test coverage like this makes sense to add only once the method is actually intrinsicified. I do not think we need to be doing future proofing for this.